### ISSUE TITLE
ruby:2.6 uses wrong bundler version

### ISSUE THREAD

**Issue by runarberg** [docker-library/ruby#252]
`rvm use 2.6` comes with bundler version 2.0.0, however `from ruby:2.6.0-rc1` installs v1.17.2. Is this intentional?

**Comment by yosifkit** [comment]
Yes and no. We install a specific version to ensure that the most up-to-date bundler is included in all the images. We did not realize that Ruby 2.6-RC was including a newer, pre-release bundler.

We'll probably need to adjust the `update.sh` script to use the newest bundler regardless of pre-release status on RC versions of Ruby.

**Comment by tianon** [comment]
Is this dependency information published anywhere?  Is the version of RubyGems and/or the version of Bundler that a particular version of Ruby comes bundled with published anywhere that we could easily scrape?

I found https://github.com/ruby/ruby/blob/v2_6_0_rc2/gems/bundled_gems, but that doesn't reference either `rubygems` nor `bundler`. :confused:

**Comment by tianon** [comment]
I guess there's https://github.com/ruby/ruby/blob/v2_6_0_rc2/lib/bundler/version.rb for `bundler`, although not exactly easy to parse (it appears `lib/bundler` is a copy of `bundler` upstream's code directly?  possibly same with `lib/rubygems`, but I'm still exploring there to try and find where the version number gets embedded)

**Comment by tianon** [comment]
https://github.com/ruby/ruby/blob/v2_6_0_rc2/lib/rubygems.rb#L12 appears to be the place the `rubygems` version gets encoded

**Comment by tianon** [comment]
(We'd probably need to use `Gem::Version` to properly parse version numbers here to determine whether a released version meets or exceeds the bundled version or whether we need to consider pre-releases, which is kind of hairy.)

**Comment by dentarg** [comment]
> whether we need to consider pre-releases

I don't think you need to consider pre-releases, [Ruby 2.6.0 has been released now](https://www.ruby-lang.org/en/news/2018/12/25/ruby-2-6-0-released/) and it uses RubyGems 3.0.1: https://github.com/ruby/ruby/blob/v2_6_0/lib/rubygems.rb#L12.
